packages/sdk/src/credential/teamsUserCredential.browser.ts

```diff
@@ -116,12 +116,24 @@ export class TeamsUserCredential implements TokenCredential {
               return;
             }
 
+            let resultJson: any = {};
             try {
-              const accessToken = parseAccessTokenFromAuthCodeTokenResponse(result);
-              resolve(accessToken);
-            } catch (error: any) {
-              reject(error);
+              resultJson = JSON.parse(result);
+            } catch (error) {
+              // If can not parse result as Json, will NOT throw error since user may return other info in auth-end page.
+              // TODO: resolve the result.
+              const failedToParseResult = "Failed to parse result to Json.";
+              internalLogger.verbose(failedToParseResult);
+              resolve();
+              return;
             }
+
+            // If sessionStorage exists in result, set the values in current session storage.
+            if (resultJson.sessionStorage) {
+              this.setSessionStorage(resultJson.sessionStorage);
```

Comment:
When will the data in session storage be used? What's the expected data returned from auth-end page?

Reply:
Offline sync: it's for msal and auth-end page will be updated later.